infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/UniqueWorkspacePVCStrategy.java

```diff
@@ -0,0 +1,113 @@
+ * of available PVCs in OpenShift project. Cleanup of backed up data is performed by removing of PVC
+ * related to the workspace.
+ *
+ * @author Anton Korneta
+ */
+public class UniqueWorkspacePVCStrategy implements WorkspacePVCStrategy {
+
+  public static final String UNIQUE_STRATEGY = "unique";
+
+  private final String pvcName;
+  private final String projectName;
+  private final String pvcQuantity;
+  private final String pvcAccessMode;
+  private final String projectsPath;
+  private final OpenShiftClientFactory clientFactory;
+
+  @Inject
+  public UniqueWorkspacePVCStrategy(
+      @Nullable @Named("che.infra.openshift.project") String projectName,
+      @Named("che.infra.openshift.pvc.name") String pvcName,
+      @Named("che.infra.openshift.pvc.quantity") String pvcQuantity,
+      @Named("che.infra.openshift.pvc.access_mode") String pvcAccessMode,
+      @Named("che.workspace.projects.storage") String projectFolderPath,
+      OpenShiftClientFactory clientFactory) {
+    this.pvcName = pvcName;
+    this.pvcQuantity = pvcQuantity;
+    this.projectName = projectName;
+    this.pvcAccessMode = pvcAccessMode;
+    this.projectsPath = projectFolderPath;
+    this.clientFactory = clientFactory;
+  }
+
+  @Override
+  public void prepare(InternalEnvironment env, OpenShiftEnvironment osEnv, String workspaceId)
+      throws InfrastructureException {
+    final String pvcUniqueName = pvcName + '-' + workspaceId;
+    final PersistentVolumeClaim pvc = osEnv.getPersistentVolumeClaims().get(pvcUniqueName);
+    if (pvc != null) {
+      return;
+    }
+    final String machineWithSources =
+        getWsAgentServerMachine(env)
+            .orElseThrow(() -> new InfrastructureException("Machine with wsagent not found"));
+    osEnv
+        .getPersistentVolumeClaims()
+        .put(pvcUniqueName, newPVC(pvcUniqueName, pvcAccessMode, pvcQuantity));
+    for (Pod pod : osEnv.getPods().values()) {
+      final PodSpec podSpec = pod.getSpec();
+      for (Container container : podSpec.getContainers()) {
+        final String machine = pod.getMetadata().getName() + '/' + container.getName();
+        if (machine.equals(machineWithSources)) {
+          final String subPath =
+              projectsPath.startsWith("/") ? projectsPath.substring(1) : projectsPath;
+          container.getVolumeMounts().add(newVolumeMount(pvcUniqueName, projectsPath, subPath));
+          podSpec.getVolumes().add(newVolume(pvcUniqueName, pvcUniqueName));
+          return;
+        }
+      }
+    }
+  }
+
+  @Override
+  public void cleanup(String workspaceId) {
+    try (OpenShiftClient client = clientFactory.create()) {
+      final String pvcUniqueName = pvcName + '-' + workspaceId;
+      final List<PersistentVolumeClaim> pvcs =
+          client.persistentVolumeClaims().inNamespace(projectName).list().getItems();
+      for (PersistentVolumeClaim pvc : pvcs) {
+        if (pvc.getMetadata().getName().equals(pvcUniqueName)) {
```

Comment:
you can get needed pvc with `client.persistentVolumeClaims().inNamespace(projectName).withName(pvcUniqueName).get();` to avoid for each for search